include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -167,7 +194,63 @@ __internal_realloc(void* __user_ptr, std::size_t __new_size)
     return __user_ptr == nullptr ? std::malloc(__new_size) : __realloc_real_pointer(__user_ptr, __new_size);
 }
 
-#endif // __linux__
+#if _WIN64
+
+inline void*
+__aligned_realloc_real_pointer(void* __user_ptr, std::size_t __new_size, std::size_t __alignment)
+{
+    assert(__user_ptr != nullptr);
+
+    if (!__new_size)
+    {
+        _aligned_free(__user_ptr);
+        return nullptr;
+    }
+
+    __block_header* __header = static_cast<__block_header*>(__user_ptr) - 1;
+
+    void* __result = nullptr;
+
+    if (__same_memory_page(__user_ptr, __header) && __header->_M_uniq_const == __uniq_type_const)
+    {
+        if (__header->_M_requested_number_of_bytes == __new_size && (uintptr_t)__user_ptr % __alignment == 0)
```

Comment:
Should not it be `_M_requested_number_of_bytes >= __new_size`?

Reply:
Yes.